x-pack/plugins/entity_manager/server/routes/entities/search.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import moment from 'moment';
+import { z } from '@kbn/zod';
+import { createEntityManagerServerRoute } from '../create_entity_manager_server_route';
+import { entitySourceSchema } from '../../lib/queries';
+
+export const searchEntitiesRoute = createEntityManagerServerRoute({
+  endpoint: 'POST /internal/entities/_search',
+  params: z.object({
+    body: z.object({
+      type: z.string(),
+      start: z
+        .optional(z.string())
+        .default(() => moment().subtract(5, 'minutes').toISOString())
+        .refine((val) => moment(val).isValid(), {
+          message: 'start should be a date in ISO format',
+        }),
+      end: z
+        .optional(z.string())
+        .default(() => moment().toISOString())
+        .refine((val) => moment(val).isValid(), {
+          message: 'start should be a date in ISO format',
+        }),
+      limit: z.optional(z.number()).default(10),
+    }),
+  }),
+  handler: async ({ request, response, params, logger, getScopedClient }) => {
+    try {
+      const { type, start, end, limit } = params.body;
+
+      const client = await getScopedClient({ request });
+
+      const sources = await client.getEntitySources({ type });
+      if (sources.length === 0) {
+        return response.notFound({ body: { message: `No sources found for type [${type}]` } });
+      }
+
+      const entities = await client.searchEntities({ sources, start, end, limit });
+
+      return response.ok({ body: { entities } });
+    } catch (e) {
+      logger.error(e);
+      return response.customError({ body: e, statusCode: 500 });
+    }
```

Comment:
Technically the router package already does this for you, so you can omit this unless you're doing some more custom error handling